app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,8 @@
+class AddressesController < APIController
+  before_action :doorkeeper_authorize!, only: [:index]
+
+  def index
+    authorize Address
+    render json: Address.all
```

Comment:
Would a walking list not be a list of addresses?

In that case, instead of `Address.all`, we would have something like `Address.walking_list_for(*something)`